osu.Game.Tests/Beatmaps/IO/ImportBeatmapTest.cs

```diff
@@ -525,6 +526,55 @@ public async Task TestImportThenDeleteThenImportWithOnlineIDsMissing()
             }
         }
 
+        [Test]
+        public void TestExportWithInvalidCharacters()
+        {
+            // unfortunately for the time being we need to reference osu.Framework.Desktop for a game host here.
+            using (HeadlessGameHost host = new CleanRunHeadlessGameHost(nameof(ImportBeatmapTest)))
+            {
+                try
+                {
+                    var osu = LoadOsuIntoHost(host);
+
+                    var metadata = new BeatmapMetadata
+                    {
+                        Artist = "Some\\Artist",
+                        AuthorString = "Some*Author",
+                        Title = "Some|Title"
+                    };
+
+                    var difficulty = new BeatmapDifficulty();
+
+                    var beatmapInfo = new BeatmapInfo
+                    {
+                        OnlineBeatmapID = 1,
+                        Metadata = metadata,
+                        BaseDifficulty = difficulty,
+                        Version = "<Some: REMIX>"
+                    };
+
+                    var beatmapToExport = new Beatmap
+                    {
+                        BeatmapInfo = beatmapInfo
+                    };
+
+                    var manager = osu.Dependencies.Get<BeatmapManager>();
+
+                    manager.Save(beatmapInfo, beatmapToExport);
+
+                    // Check that the new file is saved with valid characters
+                    Regex charRules = new Regex(@"[/\\<>|?*"":]+", RegexOptions.Multiline);
```

Comment:
Should you be using `Path.GetInvalidFileNameChars` here? We generally don't want to reinvent the wheel for cases like this.

Reply:
I'm actually not sure, because [the return value of that method may vary between platforms](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getinvalidfilenamechars?view=net-5.0#remarks).

Reply:
It should at the very least be moved to an extension as a "platform-agnostic invalid characters", bundling all invalid characters in any platform.

Reply:
Hello! Sorry for taking a long time to answer.

I've made a regex rule to check if Path.GetInvalidFileNameChars failed to remove invalid characters.

If it bypassed the function, the regex will match and it will return true, failing the test.

Reply:
I guess that's okay. The regex probably doesn't need to be multiline (and I would have done it without using a regex call but either or I guess).

Have you had a look into the test failures?